src/middlewared/middlewared/plugins/cloud_sync.py

```diff
@@ -100,8 +103,21 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 
         args += [cloud_sync["transfer_mode"].lower()]
 
+        snapshot = None
         if cloud_sync["direction"] == "PUSH":
-            args.extend([cloud_sync["path"], config.remote_path])
+            path = cloud_sync["path"]
+            if cloud_sync["snapshot"]:
```

Comment:
I had misunderstood you in the ticket.
Do you see a value for providing the feature of snapshotting out of the box?
If we have hooks the script could be as simple as:

#!/bin/sh
zfs snapshot $2@$(date +%Y%m%d%H)

I am not sure how popular having that available would be?
I also can imagine people would have request to use different snapshots names, retention policy, etc which would add the burden to us?

Reply:
Hmm, nevermind on that. I see now that you're transferring from the snapshot, I dont see how that could be accomplished with the hooks.

Reply:
> I dont see how that could be accomplished with the hooks.

We might provide some API for hook to change cloud sync task. I could not think of any other use case for that.